File: VB/WindowsApplication3/MVVM/ViewModels/DocumentsViewModel.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO

Namespace MVVMSample
    Public Class DocumentsViewModel
        Private ReadOnly viewNameSuffix As String = "View"
        Public ReadOnly Property DocumentManagerService() As IDocumentManagerService
            Get
                Return Me.GetRequiredService(Of IDocumentManagerService)()
            End Get
        End Property
        Public Overridable Property DocumentType() As DocumentType
        Protected Sub OnDocumentTypeChanged()
            LoadDocument()
        End Sub
        Public Sub LoadDocument()
            If DocumentType <> MVVMSample.DocumentType.None AndAlso DocumentManagerService IsNot Nothing Then
                Dim document As IDocument = DocumentManagerService.FindDocumentByIdOrCreate(DocumentType, Function(x) CreateDocument())
                document.Show()
            End If
        End Sub
        Private Function CreateDocument() As IDocument
            Dim document = DocumentManagerService.CreateDocument(DocumentType.ToString() & viewNameSuffix, Nothing, Me)
            document.Title = DocumentType
            document.DestroyOnClose = False
            Return document
        End Function
    End Class
    Public Enum DocumentType
        None
        Customers
        Orders
    End Enum
End Namespace

File: CS/WindowsApplication3/MVVM/ViewModels/DocumentsViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;

namespace MVVMSample {
    public class DocumentsViewModel {
        private readonly string viewNameSuffix = "View";
        public IDocumentManagerService DocumentManagerService {
            get { return this.GetRequiredService<IDocumentManagerService>(); }
        }
        public virtual DocumentType DocumentType { get; set; }
        protected void OnDocumentTypeChanged() {
            LoadDocument();
        }
        public void LoadDocument() {
            if (DocumentType != MVVMSample.DocumentType.None && DocumentManagerService != null) {
                IDocument document = DocumentManagerService.FindDocumentByIdOrCreate(DocumentType, x => CreateDocument());
                document.Show();
            }
        }
        IDocument CreateDocument() {
            var document = DocumentManagerService.CreateDocument(DocumentType.ToString() + viewNameSuffix, null, this);
            document.Title = DocumentType;
            document.DestroyOnClose = false;
            return document;
        }
    }
    public enum DocumentType {
        None,
        Customers, 
        Orders
    }
}
